spec/lib/artifact.spec.js

```diff
@@ -158,14 +158,35 @@ describe("lib/artifact", () => {
                  */
                 artifact.uri = `${uri}/MAKE_IT_FAIL`;
 
-                return artifact.downloadToFile(filename).then(() => {
-                    jasmine.fail("Did not expect things to be successful");
+                return artifact.downloadToFile(filename).then(jasmine.fail, (err) => {
+                    expect(err.code).toBe(error.NOT_FOUND);
+                });
+            });
+            it("only cleans up files when a filepath is supplied", () => {
+                var stream;
 
-                    return fs.unlinkAsync(filename);
-                }, (err) => {
+                stream = new streamBuffers.WritableStreamBuffer();
+                artifact.uri = `${uri}/MAKE_IT_FAIL`;
+
+                return artifact.downloadToFile(stream).then(jasmine.fail, (err) => {
                     expect(err.code).toBe(error.NOT_FOUND);
+                });
+            });
+            it("logs errors that occur when attempting to cleanup file", () => {
+                artifact.uri = `${uri}/MAKE_IT_FAIL`;
+                spyOn(fs, "unlinkAsync").and.returnValue(Promise.reject(new Error("no")));
 
-                    return fs.unlinkAsync(filename);
+                return artifact.downloadToFile(filename).then(jasmine.fail, (err) => {
+                    expect(err.code).toBe(error.NOT_FOUND);
+                    fs.unlinkAsync.and.callThrough();
+
+                    /* Serves dual purpose, removes the test file and blows up
+                     * if it is not there. We spied on unlink and forced it to
+                     * fail so the file should be there.
+                     */
+                    return fs.unlinkAsync(filename).then(null, (unlinkErr) => {
+                        throw unlinkErr;
```

Comment:
Could you throw a more specific error? P.S. if you didn't wan to, you could just `return fs.unlinkAsync(filename);`